### PR TITLE
[FIX] web: float field reacts to onchange with change of spec

### PR DESCRIPTION
**Pull request by SimonGenin**
The new implementation of float field has some problems dealing with
external changes that would impact it.

The specs are changed to format on blur. It was overly complicated not
to.

### CODE REVIEW

**Review by aab-odoo — COMMENTED** [review]
Commit message could explain a bit more the problem, the previous spec and the new one

File: addons/web/static/src/views/fields/input_field_hook.js [hunk 1]
```diff
@@ -18,29 +18,62 @@ export function useInputField(params) {
...
         }
     }
+
+    /**
+     * On blur, we consider the field no longer dirty, even if it were to be invalid.
+     * However, if the field is invalid, the new value will not be committed to the model.
+     */
     function onChange(ev) {
-        lastSetValue = ev.target.value;
         isDirty = false;
```
Comment (SimonGenin) [inline addons/web/static/src/views/fields/input_field_hook.js:55]: shouldn't we still check here if isDirty is false, and discard the all thing if it is ?

Comment (SimonGenin) [reply addons/web/static/src/views/fields/input_field_hook.js:55]: perhaps also the invalide state, not sure about that it's touchy. 
But it is checked in commitChanges for now.

Comment (SimonGenin) [reply addons/web/static/src/views/fields/input_field_hook.js:55]: nope

File: addons/web/static/src/views/fields/input_field_hook.js [hunk 2]
```diff
@@ -54,30 +87,63 @@ export function useInputField(params) {
...
-                component.props.setDirty(false);
+                component.props.setDirty(isDirty);
+            }
+
+            if (isInvalid) {
+                return;
+            }
+
+            if (val !== component.props.value) {
+                component.props.update(val);
```
[the remark this answers is not in the document]
Comment (SimonGenin) [reply addons/web/static/src/views/fields/input_field_hook.js:146]: yes

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**